src/state/slices/preferencesSlice/preferencesSlice.ts

```diff
@@ -90,5 +92,8 @@ export const preferences = createSlice({
     setCurrencyFormat(state, { payload }: { payload: { currencyFormat: CurrencyFormats } }) {
       state.currencyFormat = payload.currencyFormat
     },
+    setWelcomeModal(state, { payload }: { payload: { show: boolean } }) {
```

Comment:
Nitpick but `setShowWelcomeModal` would be more declarative and avoid calling `setWelcomeModal` with an object, but totally happy with this implementation